### ISSUE TITLE
PSR12.Files.FileHeader false positive when file only contains docblock

### ISSUE THREAD

**Issue by dereuromark** [squizlabs/PHP_CodeSniffer#2616]
```
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------------------------------------------
 5 | ERROR | [x] Header blocks must be followed by a single blank line
```
is unfixable.
It doesn't respect a file that only contains a docblock.

code is:
```php
<?php

/**
 * Here go your local configs
 */
```

**Comment by morrisonlevi** [comment]
I think Datadog/dd-trace-php is hitting this same issue for a different reason in [branch levi/phpcs](https://github.com/DataDog/dd-trace-php/tree/3e81033a99b09188472b2128a7ffe6c6c7211710):
```bash
circleci@82b8c67c4988:~/app$ composer lint
> phpcs -s --ignore=src/DDTrace/Integrations/ZendFramework/V1/Ddtrace.php

FILE: /home/circleci/app/tests/Unit/Util/TryCatchFinallyTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 226 | ERROR | [x] Header blocks must be followed by a single blank
     |       |     line
     |       |     (PSR12.Files.FileHeader.SpacingAfterBlock)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 39.58 secs; Memory: 20.25MB

Script phpcs -s --ignore=src/DDTrace/Integrations/ZendFramework/V1/Ddtrace.php handling the lint event returned with error code 2
circleci@82b8c67c4988:~/app$ composer fix-lint
> phpcbf

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
...cleci/app/tests/Unit/Util/TryCatchFinallyTest.php  FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 0 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 37.31 secs; Memory: 20.25MB


Script phpcbf handling the fix-lint event returned with error code 2
```

**Comment by gsherwood** [comment]
Thanks for the report. I've committed a fix, which will be in 3.5.1.

**Comment by hopeseekr** [comment]
It''s bugging out all of my repos...

We've been doing `<?php declare(strict_types=1);` for years and years. It's by far the best place for it.

**Comment by guvra** [comment]
Hi,

Is there an explanation on why it's impossible to do the following?

```php
<?php
declare(strict_types=1);
```

The blank line is a waste of space imo.

**Comment by dereuromark** [comment]
The already added newline is a big waste of space :)
There is no good reasoning here IMO - just a decision made by them that you can chose to ignore.

**Comment by gsherwood** [comment]
> Is there an explanation on why it's impossible to do the following?

Those sort of questions are better directed to the authors of PSR-12. I had nothing to do with the writing of the standard so I can't talk to the decisions within it. I've just implemented it in PHPCS as faithfully as I can.
